[TensorExpr] Add `TensorExprKernel::runFast` method.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #57329 [StaticRuntime] Use NNC's call_raw API to reduce call overheads.
* **#57328 [TensorExpr] Add `TensorExprKernel::runFast` method.**
* #55113 [TensorExpr] Add `CodeGen::call_raw` method.

This method uses `CodeGen::call_raw` instead of `CodeGen::call`.

Differential Revision: [D28110359](https://our.internmc.facebook.com/intern/diff/D28110359)

Commit:
[TensorExpr] Add `TensorExprKernel::runFast` method.

This method uses `CodeGen::call_raw` instead of `CodeGen::call`.

[ghstack-poisoned]

bertmaher commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -3149,3 +3149,20 @@ void TensorExprKernel::runKernel(Stack& stack) {
+    const std::vector<void*>& outputs) {
+  KernelScope kernelScope(&kernelArena_);
+
+  std::vector<void*> args(inputs);
```
Micro-optimization: do `args.reserve(inputs.size() + outputs.size() + constants_.size())` so we don't end up resizing this thing multiple times.

ZolotukhinM replied:
Good idea! If we want to optimize it even further, we could've returned the preallocated vector and than at the call-site we would need to only overwrite the elements corresponding to inputs/outputs pointers. But I decided not to go that far with the first version

bertmaher reviewed (approved):
rad

navahgar reviewed (approved):
LGTM

Commit:
Update on "[TensorExpr] Add `TensorExprKernel::runFast` method."


This method uses `CodeGen::call_raw` instead of `CodeGen::call`.

Differential Revision: [D28110359](https://our.internmc.facebook.com/intern/diff/D28110359)

[ghstack-poisoned]
